renovate/src/Renovate/BinaryFormat/ELF.hs

```diff
@@ -728,22 +791,22 @@ buildNewSymbolTable :: (w ~ MM.ArchAddrWidth arch, E.ElfWidthConstraints w, MM.M
                     -> RA.ConcreteAddress arch
                     -> RE.NewSymbolsMap arch
                     -> [(RA.ConcreteAddress arch, RA.ConcreteAddress arch)]
-                    -> E.ElfSymbolTable (E.ElfWordType w)
+                    -> E.Symtab w
                     -- ^ The original symbol table
-                    -> E.Elf w
-                    -> ElfRewriter lm arch (E.ElfSymbolTable (E.ElfWordType w))
-buildNewSymbolTable textSecIdx extraTextSecIdx layoutAddr newSyms addrMap baseTable elf = do
-  let newEnts = newEntries (toMap (E.elfSymbolTableEntries baseTable))
+                    -> ElfRewriter lm arch (E.Symtab w)
+buildNewSymbolTable textSecIdx extraTextSecIdx layoutAddr newSyms addrMap baseTable = do
+  let newEnts = newEntries (toMap (E.symtabEntries baseTable))
   let redirections = redirs
-  let tableEntries = V.concat [ E.elfSymbolTableEntries baseTable, V.fromList redirections, newEnts ]
-  return $ baseTable { E.elfSymbolTableEntries = tableEntries
-                     , E.elfSymbolTableIndex = nextSectionIndex elf
+  let tableEntries = V.concat [ E.symtabEntries baseTable, V.fromList redirections, newEnts ]
+  return $ baseTable { E.symtabEntries = tableEntries
+                     -- FIXME: What is this? It seems like it shouldn't be the same as the table size
```

Comment:
Should this be fixed now?  If not, should an issue be created for this?

Reply:
That is one of the last two questions I need to ask Joe